From 1fb02b4993fefa3ed522f9bae85d460cf1c6c12e Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 9 Jul 2015 09:31:10 -0700
Subject: [PATCH 18/53] dma: xilinx: dpdma: Don't set DMA_SG cap

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

The DMA_SG bit isn't for slve_sg(). Don't set the bit from
the dpdma cap mask.

Reported-by: Love Kumar <lovek@xilinx.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 08891948401be233ab887a31fddd019524fcee38)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/dma/xilinx/xilinx_dpdma.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_dpdma.c b/drivers/dma/xilinx/xilinx_dpdma.c
index 4c5e957..d8321a2 100644
--- a/drivers/dma/xilinx/xilinx_dpdma.c
+++ b/drivers/dma/xilinx/xilinx_dpdma.c
@@ -1815,7 +1815,6 @@ static int xilinx_dpdma_probe(struct platform_device *pdev)
 	INIT_LIST_HEAD(&xdev->common.channels);
 	dma_cap_set(DMA_SLAVE, ddev->cap_mask);
 	dma_cap_set(DMA_PRIVATE, ddev->cap_mask);
-	dma_cap_set(DMA_SG, ddev->cap_mask);
 	dma_cap_set(DMA_CYCLIC, ddev->cap_mask);
 	dma_cap_set(DMA_INTERLEAVE, ddev->cap_mask);
 	ddev->copy_align = fls(XILINX_DPDMA_ALIGN_BYTES - 1);
-- 
1.7.10.4

